snapshot-labs/snapshot
# feat: add subgraph queries for oov3

### Issues
Enabling oSnap creates a lot of load on web3 providers in order to query historical data. We have run into numerous limitations due to block range restrictions, parallel request restrictions, and UX responsiveness.  This PR is the start of adding subgraph support to replace the event queries.

Fixes #

### Changes 
This pr replaces oov3 event queries with the existing oov3 subgraph. We add hard coded subgraph urls in the constants file.
This also attempts to load using subgraphs first, if there is any failure, it will log and fall back to the original event based implementation, so this  should only strictly improve the situation with little risk to breaking working code.

1. adds subgraphs to contract data constants
2. adds a new getModuleDetailsUmaGql which is almost identical to the original getModuleDetailsUma function, except it surgically replaces oov3 event queries with a subgraph

### How to test
*_(Explain how the changes can be tested, including any required setup steps)_*



1. open dev console
2. go to the proposal: #/barnbridge.eth/proposal/0x74cf6f3372aaeebfc2e99ff54604a5e63ca3d44cb84bdbf202a85581dfac9f5d
3. see that it eventually loads the button "Execute transaction batch"
4. see that there is no warning in console `Error querying module details from the graph`. this means it used the subgraph function successfully

### To-Do
*_(List any outstanding tasks be addressed before or after this PR is merged)_*

- [ ] need to add OG subgraph 
- [ ] optionally refactor to reduce code duplication


### Self-review checklist

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*
